compiler/rustc_ty/src/ty.rs

```diff
@@ -276,6 +281,113 @@ fn param_env(tcx: TyCtxt<'_>, def_id: DefId) -> ty::ParamEnv<'_> {
     traits::normalize_param_env_or_error(tcx, def_id, unnormalized_env, cause)
 }
 
+fn environment<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId) -> &'tcx ty::List<Predicate<'tcx>> {
```

Comment:
can we add a doc comment to this fn? it's pretty non-obvious to me what it does from its name. Actually, I suspect the name should just be improved to something like `well_formed_types_in_env`

Reply:
```
/// Elaborate the environment.
///
/// Collect a list of `Predicate`'s used for building the `ParamEnv`.
/// 
/// Used only in chalk mode.
```
@jackh726 anything to add?

Reply:
Perhaps mention that this adds `TypeFromEnv` (or what the eventual name ends up being) predicates. Specifically, because these types come from the environment, we assume they are well formed.